packages/shared/src/components/post/PostEngagements.tsx

```diff
@@ -59,8 +60,11 @@ function PostEngagements({
     onShowShareNewComment,
   } = useShareComment(analyticsOrigin, enableShowShareNewComment);
 
-  const onCommented = (comment: Comment, isNew?: boolean) => {
+  const onCommented = (comment: Comment, isNew: boolean, parentId?: string) => {
     if (isNew) {
+      postAnalyticsEvent(AnalyticsEvent.CommentPost, post, {
+        extra: { commentId: parentId, origin: Origin.CommentModal },
```

Comment:
Please remove the origin here as discussed above.

Also commentId should then only be added if it's available

Reply:
Yes. That parameter is optional. So that value can become undefined when applicable.

Reply:
Ah yes but wouldn't this be cleaner?

```
...(parentId && { commentId: parentId })
```

That way we don't even add the prop when it's empty

Reply:
I mean, if it is undefined it is not really passed on the tracking so it would be as if there was no `commentId` prop which is the result of the suggestion.

Reply:
Are you sure, does it get send without the commentId if parentId is empty?
Or do we get commentId: undefined/null?

If the first is the case and you tested it's fine.

Reply:
Ok all good then